auth-tokens/src/main/java/com/palantir/tokens/auth/BearerToken.java

```diff
@@ -32,6 +40,47 @@
     private static final String VALIDATION_PATTERN_STRING = "^[A-Za-z0-9\\-\\._~\\+/]+=*$";
     private static final Pattern VALIDATION_PATTERN = Pattern.compile(VALIDATION_PATTERN_STRING);
 
+    private static final String SUBJECT_FIELD = "sub";
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * If the token contains a 'sub' field, as a base64 encoded UUID, return it as a String.
+     * @return the 'sub' field of the JWT, if it exists
+     */
+    public final Optional<String> getUserIdInsecure() {
```

Comment:
I think the disadvantage there is that every single consumer is going to write/use this transformation.

The question is whether this bearer token is for Palantir bearer tokens, or whether it's for generic JWTs in order to help third-party users. If we view it to be the former then we should make it return a uuid formatted string (or maybe even an actual Java UUID). If we view it to be the latter, then we should just return the 'sub' field, as you suggest.

My view is that we should do the former. This class has only been open sourced because @qinfchen wanted to use it in http-remoting, and has little value to anyone else (in particular, its value within Palantir is solely due to network effects (you don't need to convert your bearer tokens between services)). I don't think that we should compromise usage within Palantir for this.

By compromise, here I mean that I want to write

    logger.warn("User {} tried to write to {} but had no permission", bearerToken.getUserIdInsecure(), dataset);

and not

    logger.warn("User {} tried to write to {} but had no permission",
        UserIdFormatter.format(bearerToken.getUserIdInsecure()), dataset);

by which point we might as well just do

    logger.warn("User {} tried to write to {} but had no permission",
        UserIdFormatter.format(bearerToken), dataset);

The reason why I think that the latter case is worse is because UserIdFormatter is less discoverable and has more boilerplate.